nixos/nat: substitute iptables for compat under nftables

bqv opened the pull request:
###### Motivation for this change
Currently, NAT will not work if you have nftables enabled (and therefore default firewall disabled (and therefore ip_tables unloaded/blacklisted)). This allows those using nftables to enable nat and have it work with minimal code change, thanks to the good work of those who created iptables-nftables-compat.

Debugging why my containers had no network access was a pain, this change fixed my issue, I'm using it right now.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

bqv committed:
nixos/nat: substitute iptables for compat under nftables

mmilata reviewed (changes requested):
:+1: for moving towards nftables! Can we use `config.networking.firewall.package` instead of hardcoding the package name?

Also, there's another reference to `pkgs.iptables` further down in `nat.nix`, shouldn't it be replaced as well?

bqv commented:
@mmilata Thanks, updated in the new PR :D I have to give up on this one because I deleted my fork by accident and now it's unlinked, so I can't push to it
